packages/trader/src/Modules/Contract/Components/InfoBox/__tests__/info-box-longcode.spec.tsx

```diff
@@ -1,56 +1,97 @@
+            </StoreProvider>
+        );
+    };
     it('should render InfoBoxLongcode component', () => {
-        render(<InfoBoxLongcode {...mocked_props} />);
+        render(<MockInfoBoxLongcode {...mocked_props} />);
 
         expect(screen.getByText('MockedIcon')).toBeInTheDocument();
         expect(screen.getByText(test_longcode_short)).toBeInTheDocument();
     });
     it('should not render specific text if longcode is less then 150 symbols', () => {
-        render(<InfoBoxLongcode {...mocked_props} />);
+        render(<MockInfoBoxLongcode {...mocked_props} />);
 
-        expect(screen.queryByText(/View more/i)).not.toBeInTheDocument();
+        expect(screen.queryByText(view_more_text)).not.toBeInTheDocument();
     });
     it('should render specific text "View more" if longcode is more then 150 symbols', () => {
         mocked_props.contract_info.longcode = test_longcode_long;
-        render(<InfoBoxLongcode {...mocked_props} />);
+        render(<MockInfoBoxLongcode {...mocked_props} />);
 
-        expect(screen.getByText(/View more/i)).toBeInTheDocument();
+        expect(screen.getByText(view_more_text)).toBeInTheDocument();
     });
     it('should render specific text "View less" if longcode is more then 150 symbols and user click on expand button', () => {
-        render(<InfoBoxLongcode {...mocked_props} />);
-        userEvent.click(screen.getByText(/View more/i));
+        render(<MockInfoBoxLongcode {...mocked_props} />);
+        userEvent.click(screen.getByText(view_more_text));
 
-        expect(screen.queryByText(/View more/i)).not.toBeInTheDocument();
-        expect(screen.getByText(/View less/i)).toBeInTheDocument();
+        expect(screen.queryByText(view_more_text)).not.toBeInTheDocument();
+        expect(screen.getByText(view_less_text)).toBeInTheDocument();
     });
     it('should render specific text "View more" if longcode is more then 47 symbols for mobile', () => {
-        (isMobile as jest.Mock).mockReturnValue(true);
+        const mocked_store = mockStore({
+            ui: {
+                is_mobile: true,
+            },
+        });
         mocked_props.contract_info.longcode = test_longcode_mobile;
-        render(<InfoBoxLongcode {...mocked_props} />);
+        render(
+            <StoreProvider store={mocked_store}>
+                <InfoBoxLongcode {...mocked_props} />
+            </StoreProvider>
+        );
+
+        expect(screen.getByText(view_more_text)).toBeInTheDocument();
+    });
+    it('should render modal if longcode is more then 47 symbols for mobile and user clicks on "View more" button', () => {
+        const mocked_store = mockStore({
+            ui: {
+                is_mobile: true,
+            },
+        });
+        const modal_root_el = document.createElement('div');
+        modal_root_el.setAttribute('id', 'modal_root');
+        document.body.appendChild(modal_root_el);
+        mocked_props.contract_info.longcode = test_longcode_mobile;
```

Comment:
Something in Modal can be also tested like this:
```
beforeAll(() => {
        ReactDOM.createPortal = jest.fn(component => {
            return component as React.ReactPortal;
        });
    });
    afterAll(() => {
        (ReactDOM.createPortal as jest.Mock).mockClear();
    });
```
    OR just add Portal creating in one test case.
    No need in applying this suggestion, just seems easier)

Reply:
Seems more complicated to me somehow 😂